api/src/main/java/org/openmrs/ui/framework/FormatterImpl.java

```diff
@@ -113,6 +119,9 @@ private boolean wholeNumber(Number n) {
 
     private String format(Date d, Locale locale) {
         DateFormat df;
+        if(ui.handleTimeZones()){
+            return (toRFC3339(d));
+        }

```

Comment:
The "format" method is primarily about formatting something for display... are we saying when "handleTimeZones" is enabled we *never* want to format for display on the server side, but always render in RFC3339 format?

Reply:
Exactly that, because, if timezones are enable, we always need to format on the client side, with client timezone.

Reply:
Removed the UiUtils there and now we are getting the GP directly [here](https://github.com/openmrs/openmrs-module-uiframework/pull/65/files#diff-920407a71f4d88e52f65862ff99257e9eae4b910d1f050f158937182bd5b3c5eR117-R119)